### ISSUE TITLE
"try except else" block not allowed

### ISSUE THREAD

**Issue by gpotter2** [ipython/ipython#11395]
```
try:
    #...
except:
    # ...
else:
    # ...
```
blocks are not supported in IPython.

They are not really necessary, and can be worked-around, but I think it would be cleaner to support them, and it shouldn't be too complicated to achieve (simply wait for a new line as on CPython)

- How IPython does it:

![image](https://user-images.githubusercontent.com/10530980/46918853-96387200-cfd7-11e8-961e-4c34c75e0b17.png)

- How CPython does it:

![image](https://user-images.githubusercontent.com/10530980/46918870-d4ce2c80-cfd7-11e8-9e7b-caeb08b56f50.png)

Some background infos:
```
C:\Users\gpotter> python -c "import IPython; print(IPython.sys_info())"
{'commit_hash': u'a0d6ad545',
 'commit_source': 'installation',
 'default_encoding': 'cp850',
 'ipython_path': 'C:\\Python27\\lib\\site-packages\\IPython',
 'ipython_version': '5.7.0',
 'os_name': 'nt',
 'platform': 'Windows-10-10.0.18252',
 'sys_executable': 'C:\\Python27\\python.exe',
 'sys_platform': 'win32',
 'sys_version': '2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:19:30) [MSC v.1500 32 bit (Intel)]'}
```

**Comment by Carreau** [comment]
That's a bug in 7.0 that has been fixed in #11358, see #11334 we hope to release 7.1 soon. 


Be careful you seem to be mixing your ipython versions. Your first screenshot says IPython 7.0.1 and `python -c "import ..."` says 5.7. Did you mean `python3 -c ...`? 

In the meantime you can force a newline after the except block with `Ctrl-O`.

**Comment by gpotter2** [comment]
You are right on both sides, I used the wrong python version. Thanks for letting me know this was fixed !
